cli/src/stake.rs

```diff
@@ -551,18 +566,37 @@ pub fn process_create_stake_account(
         withdrawer: withdrawer.unwrap_or(config.keypair.pubkey()),
     };
 
-    let ixs = stake_instruction::create_account(
-        &config.keypair.pubkey(),
-        &stake_account_pubkey,
-        &authorized,
-        lockup,
-        lamports,
-    );
+    let ixs = if let Some(seed) = seed {
+        stake_instruction::create_account_with_seed(
+            &config.keypair.pubkey(), // from
+            &stake_account_address,   // to
+            &stake_account_pubkey,    // base
+            seed,                     // seed
+            &authorized,
+            lockup,
+            lamports,
+        )
+    } else {
+        stake_instruction::create_account(
+            &config.keypair.pubkey(),
+            &stake_account_pubkey,
+            &authorized,
+            lockup,
+            lamports,
+        )
+    };
     let (recent_blockhash, fee_calculator) = rpc_client.get_recent_blockhash()?;
+
+    let signers = if stake_account_pubkey != config.keypair.pubkey() {
+        vec![&config.keypair, stake_account] // both must sign if `from` and `to` differ
+    } else {
+        vec![&config.keypair] // when stake_account == config.keypair and there's a seed, we only need one signature
```

Comment:
@rob-solana, do you test both of these branches? Just skimming, but I would have expected 2 tests and maybe a helper to manage the shared code.

Reply:
the top case is the normal flow

the new test (in tests/stake.rs) exercises the bottom case